test/test.cpp

```diff
@@ -310,3 +310,84 @@ BOOST_AUTO_TEST_CASE(test_hasPacket_returns_false_on_internal_buffer_with_garbag
     BOOST_REQUIRE_EQUAL(4, test.readPacket(buffer, 100, 10, 1));
     BOOST_REQUIRE(!test.hasPacket());
 }
+
+BOOST_AUTO_TEST_CASE(test_open_bidirectional_udp)
+{
+    DriverTest test;
+
+    BOOST_REQUIRE_NO_THROW(test.openURI("udp://127.0.0.1:1111:2222"));
+    test.close();
+}
+
+void recv_test()
+{
+    DriverTest test;
+    uint8_t msg[4] = { 0, 'a', 'b', 0 };
+
+    test.openURI("udp://127.0.0.1:2125");
+
+    test.writePacket(msg, 4);
+    test.close();
+}
+
+BOOST_AUTO_TEST_CASE(test_recv_from_bidirectional_udp)
+{
+    DriverTest test;
+    uint8_t buffer[100];
+    int count;
+    uint8_t msg[4] = { 0, 'a', 'b', 0 };
+    boost::thread send_thread;
```

Comment:
There is absolutely no need to multithread here, you can write on one end and receive on the other in the same thread. Or am I missing something ?